mesop/components/table/table.py

```diff
@@ -1,24 +1,66 @@
-from typing import Any
+from dataclasses import dataclass
+from typing import Any, Callable
 
 import mesop.components.table.table_pb2 as table_pb
-from mesop.component_helpers import insert_component, register_native_component
+from mesop.component_helpers import (
+  insert_component,
+  register_event_handler,
+  register_event_mapper,
+  register_native_component,
+)
+from mesop.events import MesopEvent
+
+
+@dataclass(kw_only=True)
+class TableClickEvent(MesopEvent):
+  """Event representing a click on the table component cell.
+
+  Attributes:
+      row_index: DataFrame row index of the clicked cell in the table.
+      col_name: DataFrame col index of the clicked cell in the table.
+      key (str): key of the component that emitted this event.
+  """
+
+  row_index: int
+  col_index: int
+
+
+def map_table_click_event(event, key):
+  click_event = table_pb.TableClickEvent()
+  click_event.ParseFromString(event.bytes_value)
+  return TableClickEvent(
+    key=key.key,
+    row_index=click_event.row_index,
+    col_index=click_event.col_index,
+  )
+
+
+register_event_mapper(TableClickEvent, map_table_click_event)
 
 
 # Don't include type hint since Pydantic can't properly type check the Pandas data
 # frame. In addition, we don't want to include Pandas as a dependency into Mesop.
 @register_native_component
-def table(data_frame: Any):
+def table(
+  data_frame: Any, *, on_click: Callable[[TableClickEvent], Any] | None = None
+):
   """
   This function creates a table from Pandas data frame
 
   Args:
       data_frame: Pandas data frame.
+      on_click: [click](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click_event) is a native browser event.
```

Comment:
I'd mention that this is triggered whenever a table cell is clicked (similar to your comment in TableClickEvent).